### PR TITLE
Fix bug for logged-out users navigating to user profile.

### PR DESCRIPTION
**Pull request by lumapools**
Fix: Now the user can see other users' profile without being logged in.

### CODE REVIEW

**Review by bafavre — APPROVED** [review]
All good!
Just a question, should we be bale to see the email address of other users on their profile or not?

File: app/src/main/java/com/example/sharingang/UserProfileFragment.kt [hunk 1]
```diff
@@ -112,17 +112,15 @@ class UserProfileFragment : Fragment() {
 
     private fun setEmailText() {
         val emailText = binding.textEmail
-        if(isAuthUserDisplayedUser() && currentUserId != null) {
+        if(currentUserId != null && isAuthUserDisplayedUser()) {
             emailText.text = loggedInUserEmail
-        }
-        if(currentUserId != null) {
             emailText.visibility = View.VISIBLE
```
Comment (v1ruso) [inline app/src/main/java/com/example/sharingang/UserProfileFragment.kt:117]: I agree with @bafavre's comment: is the email visible to all the other users?

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -36,7 +36,7 @@
     <string name="remove_wishlist">Remove wishlist</string>
     <string name="wishlist_menu_title">Wish List</string>
     <string name="enter_address">Enter Address</string>
-    <string name="upfNotLoggedInInfo">You need to be logged in to view User Profiles.</string>
+    <string name="upfNotLoggedInInfo">You need to be logged in to view your User Profile.</string>
```
Comment (v1ruso) [inline app/src/main/res/values/strings.xml:39]: "upfNot...", I'd rename this to `userNotLoggedInInfo`or similar

**Review by v1ruso — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/sharingang/UserProfileFragment.kt (hunk 1 above)
Comment (v1ruso) [reply app/src/main/java/com/example/sharingang/UserProfileFragment.kt:116]: Indeed, which is why I wanted to make sure

**Review by v1ruso — APPROVED** [review]
Nice

**Pull request merged**